source/tutorials/hash_sampling.rst

```diff
@@ -0,0 +1,125 @@
+
+::
+
+    +-+-+-+-+-+-+      +-+      +-+     +-+      +-+-++-+-+
+    |UserRequest|  ->  |A|   -> |B|  -> |C| ->  | Database |
+    +-+-+-+-+-+-+      +-+      +-+     +-+      +-+-++-+-+
+
+
+For 3 sample user request
+    ::
+    
+    {"id": "reqid1", "data": "payload_1_original"}
+    {"id": "reqid2", "data": "payload_2_original"}
+    {"id": "reqid3", "data": "payload_3_original"}
+
+System/Service A generated logs
+    ::
+
+    {"id": "reqid1", "datetime": "<time>", "Processing_state" : "A", "Exception" : "none"}
+    {"id": "reqid2", "datetime": "<time>", "Processing_state" : "A_prime", "Exception" : "Unknow flow"}
+    {"id": "reqid3", "datetime": "<time>", "Processing_state" : "A_double_prime", "Exception" : "Parsing Failed"}
+
+System/Service B generated logs
+    ::
+    
+    {"id": "reqid1", "datetime": "<time>", "Processing_state" : "B", "Exception" : "none"}
+    {"id": "reqid2", "datetime": "<time>", "Processing_state" : "A_Failed_at_1st_Stage", "Exception" : "none"}
+    {"id": "reqid3", "datetime": "<time>", "Processing_state" : "B_Failed", "Exception" : "Field not found"}
+
+System/Service C generated logs
+    ::
+    
+    {"id": "reqid1", "datetime": "<time>", "Processing_state" : "C", "Exception" : "none"}
+    {"id": "reqid2", "datetime": "<time>", "Processing_state" : "NO_OP", "Exception" : "none"}
+    {"id": "reqid3", "datetime": "<time>", "Processing_state" : "C_Failed", "Exception" : "NullPointer"}
+
+
+With the above-generated logs on Systems/Services:-
+
+If we are doing sampling at 33% using random, we might end up forwarding below logs to logging servers.
+    ::
+    
+    {"id": "reqid1", "data": "payload_1_original"}
+    {"id": "reqid1", "datetime": "<time>", "Processing_state" : "A", "Exception" : "none"}
+    {"id": "reqid2", "datetime": "<time>", "Processing_state" : "A_Failed_at_1st_Stage", "Exception" : "none"}
+    {"id": "reqid3", "datetime": "<time>", "Processing_state" : "C_Failed", "Exception" : "NullPointer"}
+
+If we are doing mmnormalize based sampling, where we are forwarding non empty exception's field, we will forward
+    ::
+
+    {"id": "reqid2", "datetime": "<time>", "Processing_state" : "A_prime", "Exception" : "Unknow flow"}
+    {"id": "reqid3", "datetime": "<time>", "Processing_state" : "A_double_prime", "Exception" : "Parsing Failed"}
+    {"id": "reqid3", "datetime": "<time>", "Processing_state" : "B_Failed", "Exception" : "Field not found"}
+    {"id": "reqid3", "datetime": "<time>", "Processing_state" : "C_Failed", "Exception" : "NullPointer"}
+
+With these logs getting forwarded and aggregated, we will not be in a position to get end to end picture that ties the life-cycle of a request/operation in the cluster across various systems/services.
+
+
+Hash based sampling
+-------------------
+
+Using hash-based sampling we can solve this problem. In the above case if we do hash-based sampling using either of the 2 below configs
```

Comment:
Because with sampled logs we can't guarantee to get an end to end picture of the activity, 1st example above tries to explain that. Shall I add more details?

Reply:
I think your use of mmnormalize and the associated examples related to it add confusion, I would remove all references and examples of mmnormalize (except as a footnote as a way to convert text logs into JSON variables)